.ci/pitest-survival-check-xml.groovy

```diff
@@ -102,12 +93,37 @@ private static int checkPitestReport(String profile, String flag) {
     final Set<Mutation> survivingMutations = getSurvivingMutations(mutationReportNode)
 
     final File suppressionFile = new File(suppressedMutationFileUri)
-    Set<Mutation> suppressedMutations = Collections.emptySet()
+    Set<Mutation> suppressedMutations = new TreeSet<>()
     if (suppressionFile.exists()) {
         final Node suppressedMutationNode = xmlParser.parse(suppressedMutationFileUri)
         suppressedMutations = getSuppressedMutations(suppressedMutationNode)
     }
-    return compareMutations(survivingMutations, suppressedMutations, flag)
+
+    if (survivingMutations.isEmpty()) {
+        if (suppressionFile.exists()) {
+            suppressionFile.delete()
+        }
+    }
+    else {
+        final StringBuilder suppressionFileContent = new StringBuilder(1024)
+        suppressionFileContent.append(
+                '<?xml version="1.0" encoding="UTF-8"?>\n<suppressedMutations>')
+
+        survivingMutations.each {
+            suppressionFileContent.append(it.toXmlString())
+        }
+        suppressionFileContent.append('</suppressedMutations>\n')
+
+        if (suppressionFile.exists()) {
+            suppressionFile.write(suppressionFileContent.toString())
+        }
+        else {
+            suppressionFile.createNewFile()
+            suppressionFile.write(suppressionFileContent.toString())
```

Comment:
`suppressionFile.write` is the same between the 2 ifs, so that one instruction can come out.